sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -77,10 +71,44 @@ private[sql] class HiveSessionCatalog(
     metastoreCatalog.hiveDefaultTableFilePath(name)
   }
 
+  /**
+   * Get an identical copy of the `SessionCatalog`.
+   * The temporary tables and function registry are retained.
+   * The table relation cache will not be populated.
+   * @note `externalCatalog` and `globalTempViewManager` are from shared state, don't need deep copy
+   * `FunctionResourceLoader` is effectively stateless, also does not need deep copy.
+   * All arguments passed in should be associated with `sparkSession`.
+   * This should ideally override `SessionCatalog.copy` but does not at present, since
+   * `HiveMetastoreCatalog` is dependent on `SparkSession`.
+   */
+  def clone(
```

Comment:
This should ideally override `SessionCatalog.clone` but does not at present, since `HiveMetastoreCatalog` is dependent on `SparkSession`.